auth.business/src/main/java/com/github/pplociennik/auth/business/authentication/AuthenticationFacade.java

```diff
@@ -101,6 +111,29 @@ public void confirmRegistration( @NonNull String aToken ) {
         publishEventOnAccountConfirmationFinished( enabledAccount );
     }
 
+    /**
+     * Validates login data and authenticates user.
+     *
+     * @param aLoginDO
+     *         a login data.
+     */
+    public void authenticateAccount( @NonNull LoginDO aLoginDO ) {
+        requireNonNull( aLoginDO );
+        //validationService.validateLogin( aLoginDO );
+        var account = getAccountByUsernameOrEmail( aLoginDO.getUsernameOrEmail() );
+        var token = new UsernamePasswordAuthenticationToken( account.getUsername(), account.getPassword() );
+        var authenticationObject = authenticationManager.authenticate( token );
+        SecurityContextHolder
+                .getContext()
+                .setAuthentication( authenticationObject );
+    }
+
+    private AccountDO getAccountByUsernameOrEmail( String aUsernameOrEmail ) {
+        return aUsernameOrEmail.contains( "@" )
+               ? accountRepository.findAccountByEmailAddress( aUsernameOrEmail )
+               : accountRepository.findAccountByUsername( aUsernameOrEmail );
+    }
```

Comment:
Jest zdefiniowany regex dla maili w validatorze uwierzytelniania. Dużo lepiej będzie to wyglądać niż takie bieda-rozwiązanie

Reply:
done